[Flutter-Parent] Upgrade dependencies and update gitignore

StephenBrough opened the pull request:
Upgrading Flutter to the newest stable requires a newer version of the intl dependency. I also ran a package upgrade to our dependencies.

I added `.flutter-plugins-dependencies` to the gitignore as well.

instructure-cx commented:
Scan submitted to Checkmarx

instructure-cx commented:
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1014878&projectid=20)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|0
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

CalvinKern commented on apps/flutter_parent/test/api/api_prefs_test.dart:
```diff
@@ -18,14 +18,16 @@ import 'package:flutter_parent/api/utils/api_prefs.dart';
 
 void main() {
+  TestWidgetsFlutterBinding.ensureInitialized();
+
```
Why is this necessary? This test doesn't even use widgets.

CalvinKern replied:
Hmmm, I still don't know why this changed to require this, but looks like anything that calls `setupPlatformChannels` will need to do this. We may want to add it there instead of having each test run this line that doesn't use a widget test.

StephenBrough replied:
Good idea, I'll move it over there so we don't have to put it everywhere

CalvinKern commented:
Were all these tests passing locally when you were running them? It would be highly concerning if bitrise tests fail more than our local machines; doing a bunch of "test fix" commits would be terrible to have to do long term.

StephenBrough commented:
> Were all these tests passing locally when you were running them? It would be highly concerning if bitrise tests fail more than our local machines; doing a bunch of "test fix" commits would be terrible to have to do long term.

Yep, everything seems to be passing for me locally. I'm even seeing issues with tests that were using Future errors now that didn't have issues before. Maybe there's something going on with our Bitrise config that's causing issues.

CalvinKern commented:
>Yep, everything seems to be passing for me locally. I'm even seeing issues with tests that were using Future errors now that didn't have issues before. Maybe there's something going on with our Bitrise config that's causing issues.

😢 

That's very unfortunate. I'll take a look at the config and see if I can't find the disconnect.

StephenBrough commented:
> > Yep, everything seems to be passing for me locally. I'm even seeing issues with tests that were using Future errors now that didn't have issues before. Maybe there's something going on with our Bitrise config that's causing issues.
> 
> 😢
> 
> That's very unfortunate. I'll take a look at the config and see if I can't find the disconnect.

It may have been some weird caching issue on my end. I just cleaned, reverted, and ran them again. Some of them failed where they passed before.
